tests/trx_generator/trx_generator_tests.cpp

```diff
@@ -435,4 +483,71 @@ BOOST_AUTO_TEST_CASE(account_name_generator_tests)
    }
 }
 
+BOOST_AUTO_TEST_CASE(simple_http_client_async_test) {
+
+   auto const host     = "127.0.0.1";
+   auto const port     = 8888;
+   auto const port_str = "8888";
+
+   // Start Server
+   trx_gen_impl                   server = trx_gen_impl();
+   auto                           addr   = boost::asio::ip::address::from_string(host);
+   boost::asio::ip::tcp::endpoint endpoint(addr, port);
+
+   server.start(endpoint);
+
+   // Start Client
+
+   // The io_context is required for all I/O
+   net::io_context ioc;
+   auto const      target        = "/v1/chain/test";
+   int             version       = 11;
+   auto const      content_type  = "text/plain";
+   auto const      content_type2 = "application/json";
```

Comment:
Do you specify `const` to mean the pointer cannot be changed? `auto x = "string";` is already a `const char *`.

And why `int` for version instead of auto? `auto version = 11;` is an int by default.

I think these could all be `constexpr auto`. And I would make the strings be `std::string` by adding the `s` suffix, so that there isn't a conversion to `std::string` for every comparison, which is what I think occurs.

Reply:
`constexpr` for `string_literals` isn't available until C++20. So opted for a different direction here.

Addressed: [f02287e](https://github.com/AntelopeIO/leap/pull/1052/commits/f02287e44535e51b7ae21d1c90db73c40f51d644)